### ISSUE TITLE
PyMuPDF over Anaconda

### ISSUE THREAD

**Issue by klatifch** [rk700/PyMuPDF#206]
How can I install PyMyPDF on anaconda? Can't find anything in defaults, anaconda-fusion, or conda-forge.

**Comment by JorjMcKie** [comment]
Guesss I don't understand you well: You mean Anaconda prevents you from using PIP?

**Comment by klatifch** [comment]
One can install using pip for sure. But then, can't share or port the envoirnment "as is" from dev to production. Would be great if the package can be installed using conda.

**Comment by JorjMcKie** [comment]
I have no experience with Anaconda.
I am using plain Python installs from Python.org because I have to support 10 Python installations in parallel. I have no intention to install Anaconda, PyCharm or whatever else of those integrated environments.
The installation itself for every environment is very, very simple: 3 Python sources and one binary packed in a wheel.
So it would be great if you could help out here by providing a script or whatever on how to install in an Anaconda environment.

**Comment by maximlt** [comment]
It would actually be really nice to have the package available on conda-forge for instance.
Processing PDF files and images is something that is done quite often in the context of data analysis, conda being very much used in that area. Of course it's possible to `pip install` it, but it's as nice and may end up in getting broken environments.

**Comment by JorjMcKie** [comment]
Ok, ok, @maximlt & @klatifch, I can see you are insisting.
What must be done to make this happen? I have no time to do a lot of R&D.

**Comment by maximlt** [comment]
To be honest, I don't really know. I may have a look at it, not in the coming days unfortunately.
I just wanted to point out that I'm not really insisting. I can pip install your package and that works fine! The thing is, I know that sometimes if I can't find a package on a conda channel, I'll just try to find an alternative one instead of using pip. When it comes to image and pdf processings, there are a lot of alternatives! Having your package on conda could attract some users, which would be good for you in the end I guess :)

**Comment by yig** [comment]
In case anyone wants to take this on, here are instructions for adding a Python package to conda-forge: https://conda-forge.org/docs/maintainer/adding_pkgs.html

**Comment by JorjMcKie** [comment]
@yig thanks a lot for this hint!

**Comment by wolfv** [comment]
Hi @JorjMcKie I'd be happy to help. Usually, the process starts with adding a package to staged recipes. We could try to link to the already existing mupdf (is version 1.17.0 recent enough?)

https://github.com/conda-forge/mupdf-feedstock

**Comment by JorjMcKie** [comment]
Hi @wolfv - thank you very much for your willingness to help!

There are basically two ways to install: via wheels or via generating from sources.
* the wheels are prepared by Tracis CI (for Linux and Mac OSX) and on my own computer for Windows. They are self-sufficient and need no previously installed MuPDF at all. There exist 64bit wheels for Python versions 3.6 through 3.9. Support for earlier versions has been abandoned (for both, wheels as well as for source builds). For Windows I am also providing 32bit versions btw.
* Generating from sources **_does_** need an installed MuPDF. As I am keeping the version number of PyMuPDF (major an minor) in sync with the one of MuPDF, the current PyMuPDF - having a version 1.18.8 - needs MuPDF v1.18.0.

**Comment by JorjMcKie** [comment]
As I wrote in another thread on the same topic:
I need help for this! I do not use conda and have no experience with it.
So far nobody volunteered, so I guess nothing will happen here as long as this does not change.
